### PR TITLE
Add option for aks node restriction

### PR DESCRIPTION
**Pull request by lht6**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create --enable-node-restriction

az aks update --enable-node-restriction
az aks update --disable-node-restriction

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -13,6 +13,13 @@ Pending
 +++++++
 *  [BREAKING CHANGE] Since the service no longer supports updating source resource id for role binding, so remove --source-resource-id of `aks trustedaccess rolebinding update` command
 
+0.5.94
+++++++
+
+* Add --enable-node-restriction
+* Update to use 2022-06-02-preview api version.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:20]: This was done in 0.5.92

Comment (FumingZhang) [reply src/aks-preview/HISTORY.rst:20]: Please also move the items under the `Pending` section to this release.

**Review by FumingZhang — COMMENTED** [review]
The PR looks good to me, could you please add a live test case for it?

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -260,6 +260,39 @@ def test_aks_create_and_update_with_managed_aad_enable_azure_rbac(self, resource
             self.check('aadProfile.enableAzureRbac', False)
         ])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_create_and_update_with_node_restriction(self, resource_group, resource_group_location):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:265]: Queued a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=59472133&view=results) to generate the recording file for this newly added test case.

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:265]: Queued a new [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=59726478&view=results), it passed.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**